app/components/DataInput/index.js

```diff
@@ -47,21 +46,16 @@ class DataInput extends AppComponent {
     ];
   }
 
-  componentWillMount() {
-    this.setState({ rawData: this.props.rawData });
-  }
-
   componentWillReceiveProps(nextProps) {
     if (this.props.rawData !== nextProps.rawData) {
       this.setState({ rawData: nextProps.rawData });
     }
   }
 
   _submitData(data) {
-    // setState is async so we apply trim() twice
-    this.setState({ rawData: data.trim() });
-    this.props.dispatch(
-      actionTrigger(RECEIVE_RAW_DATA, data.trim())
+    const rawData = data.trim();
+    this.setState({ rawData }, () =>
```

Comment:
I change this, because of the comment that setState was async.

As far as an answer to your question, I believe the reason is because the "data" isn't saved to the store until the user clicks off of the text area, while the data in the text area can be changed in the mean time. In other words, onBlur vs onChange.

I think a lot of things can be simplified as far as state management, but that may be something that needs to be handled in a larger/more concentrated refactoring effort.